src/plugins/template/src/infer_request.cpp

```diff
@@ -0,0 +1,253 @@
+}
+// ! [infer_request:infer_preprocess]
+
+// ! [infer_request:start_pipeline]
+void TemplatePlugin::InferRequest::start_pipeline() {
+    OV_ITT_SCOPED_TASK(itt::domains::TemplatePlugin, m_profiling_task[StartPipeline])
+    auto start = Time::now();
+    m_executable->call(m_plugin_output_tensors, m_plugin_input_tensors);
+    m_durations[StartPipeline] = Time::now() - start;
+}
+// ! [infer_request:start_pipeline]
+
+void TemplatePlugin::InferRequest::wait_pipeline() {
+    OV_ITT_SCOPED_TASK(itt::domains::TemplatePlugin, m_profiling_task[WaitPipeline])
+    auto start = Time::now();
+    // TODO: Wait pipeline using driver API or other synchronizations methods
+    // NOTE: not used in current implementation since `startPipeline` executes pipiline synchronously
+    m_durations[WaitPipeline] = Time::now() - start;
+}
+
+// ! [infer_request:infer_postprocess]
+void TemplatePlugin::InferRequest::infer_postprocess() {
+    OV_ITT_SCOPED_TASK(itt::domains::TemplatePlugin, m_profiling_task[Postprocess]);
+    auto start = Time::now();
+    // We don't need post-processing in template plugin
+    m_durations[Postprocess] = Time::now() - start;
+}
+// ! [infer_request:infer_postprocess]
+
+// ! [infer_request:get_blob]
+ov::Tensor TemplatePlugin::InferRequest::get_tensor(const ov::Output<const ov::Node>& port) const {
+    OV_ITT_SCOPED_TASK(itt::domains::TemplatePlugin, "get_tensor");
+    auto found_port = find_port(port);
+    OPENVINO_ASSERT(found_port.found(), "Failed to get tensor, port ", port, " cannot be found.");
+
+    ov::Tensor data;
+    if (found_port.is_input()) {
+        data = m_input_tensors[found_port.idx];
+        ov::Shape shape;
+        if (!data.get_size()) {
+            auto&& parameters = get_template_model()->m_model->get_parameters();
+            const auto& pshape = parameters.at(found_port.idx)->get_partial_shape();
+            shape = pshape.is_dynamic() ? ov::Shape{0} : pshape.get_shape();
+            auto this_non_const = const_cast<InferRequest*>(this);
+            AllocateImplSingle(this_non_const->m_input_tensors, found_port.idx, port.get_element_type(), shape);
+            data = m_input_tensors[found_port.idx];
+        } else {
+            shape = data.get_shape();
+        }
+        check_tensor(port, data);
+    } else {
+        data = m_output_tensors[found_port.idx];
+        ov::Shape shape;
+        auto has_zeros = [](const ov::Shape& vec) {
+            return std::any_of(vec.cbegin(), vec.cend(), [](size_t e) {
+                return e == 0;
+            });
+        };
+        if (port.get_partial_shape().is_static() && !has_zeros(port.get_shape())) {
+            shape = port.get_shape();
+        } else if (!has_zeros(data.get_shape())) {
+            shape = data.get_shape();
+        } else {
+            shape = ov::Shape(
+                port.get_partial_shape().rank().is_dynamic() ? 1 : port.get_partial_shape().rank().get_length(),
+                0);
+        }
```

Comment:
looks like the code above is not template plugin specific, should we think of creation of a dedicated helper / method somewhere in nrgaph?